File: hackerrank/java/30-days-of-code/day-23-bst-level-order-traversal.java
import java.io.*;
import java.util.*;
class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data=data;
        left=right=null;
    }
}
class Solution{

	static void levelOrder(Node root){
        Queue<Node> q = new LinkedList<>(); 
        Queue<Integer> data = new LinkedList<>();
        Node temp = root;

        q.add(temp); 

        while (!q.isEmpty()) {
            temp = q.remove();
            data.add(temp.data);
            if (temp.left != null) {
                q.add(temp.left);
            } 
            if (temp.right != null) {
                q.add(temp.right);
            }
        }
        for (int num : data) {
            System.out.print(num + " ");
        }
      
    }

	public static Node insert(Node root,int data){

File: hackerrank/python/30-days-of-code/day-23-bst-level-order-traversal.py
class Node:
    def __init__(self,data):
        self.right=self.left=None
        self.data = data
class Solution:
    def insert(self,root,data):
        if root==None:
            return Node(data)
        else:
            if data<=root.data:
                cur=self.insert(root.left,data)
                root.left=cur
            else:
                cur=self.insert(root.right,data)
                root.right=cur
        return root

    def levelOrder(self,root):
        queue = []
        data = []
        temp = root

        queue.append(temp)

        while len(queue):
            temp = queue.pop(0)
            data.append(temp.data)
            if temp.left:
                queue.append(temp.left)
            if temp.right:
                queue.append(temp.right)

        for int in data:
            print(str(int), end=" ")